### ISSUE TITLE
Regression in scipy.integrate.dblquad: integration fails with extra arguments

### ISSUE THREAD

**Issue by EmperorDali** [scipy/scipy#6571]
There appears to be a regression in `scipy.integrate.dblquad`:

```
from scipy.integrate import dblquad
f = lambda y,x,a: a*x*y
dblquad(f, 0, 1, lambda y: 0, lambda y: 1, args=(1,))
```
works in 0.17.1, but in 0.18 causes the error:

```
.../python3.4/site-packages/scipy/integrate/quadpack.py in integrate(self, *args, **kwargs)
    747         ind = -(depth + 1)
    748         fn_range = self.ranges[ind]
--> 749         low, high = fn_range(*args)
    750         fn_opt = self.opts[ind]
    751         opt = dict(fn_opt(*args))

TypeError: <lambda>() takes 1 positional argument but 2 were given
```
Changing the call to

`dblquad(lambda y,x,a: 1, 0, 1, lambda x,a: 0, lambda x,a: 1,args=(1,))`

does not fix the problem. Version info:

Python 3.4.3 (default, Oct 14 2015, 20:28:29)
Type "copyright", "credits" or "license" for more information.

IPython 5.0.0 -- An enhanced Interactive Python.

In [3]: print( scipy.__version__)
0.18.0

**Comment by ev-br** [comment]
Duplicate of https://github.com/scipy/scipy/issues/6458? Can you please check the master branch, it should be fixed in it.

**Comment by ev-br** [comment]
And indeed, in the branch of https://github.com/scipy/scipy/pull/6557,

```
In [4]: f = lambda y,x,a: a*x*y

In [5]: dblquad(f, 0, 1, lambda y: 0, lambda y: 1, args=(1,))
Out[5]: (0.24999999999999997, 5.539061329123429e-15)
```

so this should be fixed in master and the fix will be available in scipy 0.18.1 when it's released.

**Comment by ev-br** [comment]
Erm, I did not say it explicitly: thanks for reporting the issue!
